x/staking/types/validator_test.go

```diff
@@ -319,6 +319,11 @@ func TestBondStatus(t *testing.T) {
 	require.Equal(t, types.BondStatusUnbonding, types.Unbonding.String())
 }
 
+func TestValidatorSetConsensusAddress(t *testing.T) {
+	validator := newValidator(t, valAddr1, pk1)
+	require.Equal(t, sdk.ConsAddress(pk1.Address()).String(), validator.ConsensusAddress)
```

Comment:
Not sure but it seems to me that the ConsAddress.String method is calling the same code as testcodec. To be on the safe side, it would be great to just have the bech32 encoded string here instead

Reply:
The problem is that pk1 is a random key, that's why we did that